src/wire/ipv6ext.rs

```diff
@@ -0,0 +1,246 @@
+
+    /// Return the option type.
+    #[inline]
+    pub fn option_type(&self) -> OptionType {
+        let data = self.buffer.as_ref();
+        OptionType::from(data[field::TYPE])
+    }
+
+    /// Return the length of the data.
+    ///
+    /// # Panics
+    /// The function panics if the type does not support this field
+    #[inline]
+    pub fn option_data_length(&self) -> u8 {
+        let data = self.buffer.as_ref();
+        data[field::LENGTH]
+    }
+
+    /// Return the option data.
+    ///
+    /// # Panics
+    /// The function panics if the type does not support this field
+    #[inline]
+    pub fn option_data(&self) -> &[u8] {
+        let data = self.buffer.as_ref();
+        let len = (self.option_data_length() as usize) + field::DATA.start;
+        &data[field::DATA.start..len]
+    }
+}
+
+impl<T: AsRef<[u8]> + AsMut<[u8]>> Packet<T> {
+    /// Set the option type.
+    #[inline]
+    pub fn set_option_type(&mut self, value: OptionType) {
+        let data = self.buffer.as_mut();
+        data[field::TYPE] = value.into();
+    }
+
+    /// Set the option data and data length
+    ///
+    /// # Panics
+    /// The function panics if the type does not support the length or data fields
+    #[inline]
+    pub fn set_option_data(&mut self, value: &[u8], len: u8) {
+        let data = self.buffer.as_mut();
+        data[field::LENGTH] = len;
+
+        let len = len as usize;
+
+        // TODO find idiomatic way of doing this
+        //data[field::DATA.start..len] = value[..len];
+        for i in field::DATA.start..len {
+            data[i] = value[(i-2)];
+        }
+    }
+}
+
+/// A high-level representation of an IPv6 Extension Header Option
+#[derive(Debug, PartialEq, Eq, Clone, Copy)]
+pub enum Repr<'a> {
+    Pad1 {
+        ident:  OptionType,
+    },
+    PadN {
+        ident:  OptionType,
+        length: u8, // TODO -  should this value represent the total length or the data length?
+        data:   &'a [u8]
+    },
+
+    #[doc(hidden)]
+    __Nonexhaustive
+}
+
+impl<'a> Repr<'a> {
+    /// Parse an IPv6 Extension Header Option packet and return a high-level representation.
+    pub fn parse<T>(packet: &'a Packet<&'a T>) -> Result<Repr<'a>> where T: AsRef<[u8]> + ?Sized {
+        match packet.option_type() {
+            OptionType::Pad1 => {
+                Ok(Repr::Pad1 {
+                    ident: packet.option_type(),
+                })
+            }
+            OptionType::PadN => {
+                Ok(Repr::PadN {
+                    ident: packet.option_type(),
+                    length: packet.option_data_length(),
+                    data: packet.option_data(),
+                })
+            }
+            _ => Err(Error::Unrecognized),
+        }
+    }
+
+    /// Return the length of a header that will be emitted from this high-level representation.
+    pub fn buffer_len(&self) -> usize {
+        // This function is not strictly necessary, but it can make client code more readable.
```

Comment:
I'd remove this comment. This is the case for headers like `ipv6` where the buffer length is static.

Reply:
removed